spec.html

```diff
@@ -28657,8 +28657,6 @@ <h1>Runtime Semantics: GetSubstitution ( _matched_, _str_, _position_, _captures
             1. Assert: Type(_replacement_) is String.
             1. Let _tailPos_ be _position_ + _matchLength_.
             1. Let _m_ be the number of elements in _captures_.
-            1. If _namedCaptures_ is not *undefined*, then
-              1. Set _namedCaptures_ to ? ToObject(_namedCaptures_).
```

Comment:
hmm - if `functionalReplace` is false, and `namedCaptures` is null, wouldn't this change be observable?

Reply:
These are the steps between current and proposed location of `ToObject`:

```
1. Assert: Type(_matched_) is String.
1. Let _matchLength_ be the number of code units in _matched_.
1. Assert: Type(_str_) is String.
1. Let _stringLength_ be the number of code units in _str_.
1. Assert: _position_ is a nonnegative integer.
1. Assert: _position_ &le; _stringLength_.
1. Assert: _captures_ is a possibly empty List of Strings.
1. Assert: Type(_replacement_) is String.
1. Let _tailPos_ be _position_ + _matchLength_.
1. Let _m_ be the number of elements in _captures_.
```

None of them seems observable, but I'd rather have throwing exception before all the asserts.